enterprise/coderd/licenses.go

```diff
@@ -208,25 +119,25 @@ func (a *licenseAPI) postLicense(rw http.ResponseWriter, r *http.Request) {
 		})
 		return
 	}
-	err = a.pubsub.Publish(PubSubEventLicenses, []byte("add"))
+	err = api.updateEntitlements(r.Context())
```

Comment:
this is unnecessary; the publishing to the pubsub triggers the update.

Reply:
I'm guessing this comment:

> Is it fair to log this error instead then? It seems awkward to add a license, then potentially have a subsequent request not have the licensed functionality. If I'm automating Coder, would my route be to poll and wait for the pubsub to have triggered new entitlements?

was for this instead.

Unfortunately, triggering the entitlements update within the `postLicense()` handler only ensures that this particular coderd instance has processed the update.  As soon as you have a load balancer and multiple coderds, then we are back to the same problem of subsequent requests racing against the processing (in the general case, anyway.  If you wrote code that re-used the same TCP session then it is unlikely you would get routed to a different coderd, but that's much more involved than we expect people automating coder will do.)

What I would recommend is that people just prepare to retry any requests that use licensed features that immediately follow a license add.  Polling, unfortunately, suffers the same problem where it's hard to guarantee that the coderd that responded to your poll is also the one that will respond to your next request.

Reply:
That's a really good point... in an HA setup, they couldn't depend on it anyways. Still, it feels silly to not synchronously process an update to ensure a license was registered successfully rather than asynchronously handle it and log the error.

I propose we do both and leave it as-is for now. It kinda covers the best of both worlds (even if the entitlements will be getting updated twice, which is definitely kinda bleh).